dss/stepfunctions/gscopyclient/implementation.py

```diff
@@ -139,6 +139,19 @@ def _sfn():
                 "Resource": copy_worker,
                 "Retry": _retry_default(),
                 "Catch": _catch_default(),
+                "Next": "TestFinished",
+            },
+            f"TestFinished": {
+                "Type": "Choice",
+                "Choices": [{
+                    "Variable": "$.finished",
```

Comment:
please excuse this dumb question: if the variable value is not marked as "true" what mechanism saves the context of the state so it's not lost?

Reply:
The state is saved by the AWS step function (SFN) execution engine.

The SFN calls Lambda functions on some state and saves the state returned. The [SFN language](https://docs.aws.amazon.com/step-functions/latest/dg/concepts-amazon-states-language.html) also provides some flow control steps, such as the `Choice` step used here.